span.go

```diff
@@ -397,6 +397,10 @@ func (s *Span) End() {
 
 		if s.Type != parentType || s.Subtype != parentSubtype {
 			s.dropWhen(true)
+			if s.tx != nil {
+				s.tx.mu.Lock()
+				defer s.tx.mu.Unlock()
```

Comment:
Reading the godoc in `span.end()`, `s.tx.transactionData.mu` also needs to be held:

https://github.com/elastic/apm-agent-go/blob/326e364d6e52ea9601deb41a0004b633e1a03051/span.go#L466-L471

```suggestion
				defer s.tx.mu.Unlock()
				if !s.tx.ended() {
					s.tx.TransactionData.mu.Lock()
					defer s.tx.TransactionData.mu.Unlock()
				}
```